src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexRunnerFactory.cs

```diff
@@ -101,7 +103,23 @@ protected override void Go()
                     // If we successfully matched, capture the match, and then jump the current position to the end of the match.
                     int start = pos.Index + beginning;
                     int end = start + pos.Length;
-                    Capture(0, start, end);
+                    if (pos.CaptureStarts != null)
+                    {
+                        Debug.Assert(pos.CaptureEnds != null);
+                        Debug.Assert(pos.CaptureStarts.Length == pos.CaptureEnds.Length);
+                        for (int cap = 0; cap < pos.CaptureStarts.Length; ++cap)
+                        {
+                            if (pos.CaptureStarts[cap] >= 0)
+                            {
+                                Debug.Assert(pos.CaptureEnds[cap] >= 0);
```

Comment:
Assert pos.CaptureEnds[cap] >= pos.CaptureStarts[cap] ?

Reply:
I'll add that.